pkg/controller/generic_control.go

```diff
@@ -263,6 +267,23 @@ func (w *typedWrapper) CreateOrUpdateConfigMap(controller client.Object, cm *cor
 	return result.(*corev1.ConfigMap), nil
 }
 
+func (w *typedWrapper) CreateOrUpdateTidbInitializer(controller client.Object, tidbInitializer *v1alpha1.TidbInitializer) (*v1alpha1.TidbInitializer, error) {
```

Comment:
Please do not use the create-or-update function anymore, we have several PRs to remove the usage of the similar functions, which will call the kube-apiserver at least once during each sync.
BTW, no need to update the initializer, just create if not exist.